addons/nuxeo-ai-aws-core/src/main/java/org/nuxeo/ai/endpoint/Rekognition.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * (C) Copyright 2006-2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ * Contributors:
+ *     anechaev
+ */
+package org.nuxeo.ai.endpoint;
+
+import static java.util.Collections.singletonMap;
+import static org.nuxeo.ai.rekognition.listeners.AsyncLabelResultListener.JOB_ID_CTX_KEY;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.nio.charset.Charset;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+
+import org.apache.commons.io.IOUtils;
+import org.nuxeo.ai.pojo.Notification;
+import org.nuxeo.ai.rekognition.listeners.AsyncCelebritiesResultListener;
+import org.nuxeo.ai.rekognition.listeners.AsyncFaceResultListener;
+import org.nuxeo.ai.rekognition.listeners.AsyncLabelResultListener;
+import org.nuxeo.ai.rekognition.listeners.AsyncUnsafeResultListener;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.core.event.EventService;
+import org.nuxeo.ecm.core.event.impl.EventContextImpl;
+import org.nuxeo.ecm.webengine.model.WebObject;
+import org.nuxeo.runtime.api.Framework;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+@Path("/ai/rekognition")
+@WebObject(type = "rekognition")
+public class Rekognition {
+
+    protected static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    @POST
+    @Path("labels")
+    @Consumes({"application/json"})
+    public Response callback(@Context HttpServletRequest request) throws IOException {
+        String json = IOUtils.toString(request.getInputStream(), Charset.defaultCharset());
+        Notification notification = OBJECT_MAPPER.readValue(json, Notification.class);
+
+
+        EventContextImpl ctx = new EventContextImpl();
+        Map<String, Serializable> props = singletonMap(JOB_ID_CTX_KEY, notification.getJobId());
+        ctx.setProperties(props);
+
+        boolean succeeded = notification.isSucceeded();
+
+        String event;
+        switch (notification.getApi()) {
```

Comment:
So you have a generic handler for different calls to Rekognition.  You should explain that in the javadoc - say what's supported.  I count 5 services in the table here: https://docs.aws.amazon.com/rekognition/latest/dg/video.html